### PR TITLE
108324 extend reject to send mail

### PR DESCRIPTION
**Pull request by janid1967**
[AB#108324](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/108324)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Command/Email/CompletionMailService.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        var mailTemplate = await _mailTemplateRepository.GetByCodeAsync(_plantProvider.Plant, eMailCode, cancellationToken);
+
+        var subject = _templateTransformer.Transform(mailTemplate.Subject, context);
+        var body = _templateTransformer.Transform(mailTemplate.Body, context);
+
+        if (_options.CurrentValue.FakeEmail)
+        {
+            var fakeBody = "This email is sent to you only since fake email is enabled. " +
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/Email/CompletionMailService.cs:61]: Use string interpolation? 
@$

File: src/Equinor.ProCoSys.Completion.WebApi/appsettings.json [hunk 2]
```diff
@@ -43,6 +45,14 @@
     "ServiceBus": "",
     "AppConfig": ""
   },
+  "Email": {
+    "MailUserOid": "70f02efd-ba53-498c-8a75-09d2d7f31d61"
+  },
+  "Graph": {
+    "TenantId": "3aa4a235-b6e2-48d5-9195-7fcf05b459b0",
+    "ClientId": "0ae66748-e5b8-418b-8f4f-291c368312df",
+    "ClientSecret": "Npz8Q***************"
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.WebApi/appsettings.json:54]: Having client ids / oids and first 5 characters in client secret is okay to check into a public repository?
Just checking...

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/Email/CompletionMailServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        await _dut.SendEmailAsync(_context, _code, _eMailAddresses, CancellationToken.None);
+
+        // Assert
+        await _emailServiceMock.Received(1).SendEmailsAsync(
+            _eMailAddresses, 
+            _transformedSubject, 
+            _transformedBody, 
+            Arg.Any<CancellationToken>());
+    }
+
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.Command.Tests/Email/CompletionMailServiceTests.cs:83]: Do we need a test for when _emailAddresses is empty, then should be no call to SendEmailsAsync?

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs [hunk 4]
```diff
@@ -60,7 +67,7 @@ public async Task<Result<string>> Handle(RejectPunchItemCommand request, Cancell
 
         try
         {
```
Comment (trondlindanger) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs:69]: Quite broad try-catch block. This can catch a wide range of exceptions. Do we want to split it into finer-grained ones?

**Review by trondlindanger — APPROVED** [review]

File: src/Equinor.ProCoSys.Completion.WebApi/appsettings.json (hunk 2 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.WebApi/appsettings.json:54]: I've asked our security champion if this is a breach. I'll replace with "" so we don't do it again

File: src/Equinor.ProCoSys.Completion.Command/Email/CompletionMailService.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/Email/CompletionMailService.cs:61]: I'll keep it as is

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/Email/CompletionMailServiceTests.cs (hunk 3 above)
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.Command.Tests/Email/CompletionMailServiceTests.cs:83]: added

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs (hunk 4 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs:69]: I don't see that as an improvement. The callstack will be logged
